File: ThreadingSimulator/Models/LogModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThreadingSimulator.Enums;

namespace ThreadingSimulator.Models
{
    public class LogModel : BaseLogModel
    {
        public int Process { get; set; }
        public CommandModel Command { get; set; }
        
        public bool ShouldDisplay
        {
            get
            {
                return Command != null;
            }
        }
    }
}

File: GUI/src/app/Models/LogModel.ts
import { BaseLogModel } from "./BaseLogModel";
import { CommandModel } from "./CommandModel";

export class LogModel extends BaseLogModel
{
    public Process: number;
    public Command: CommandModel;
    
    public ShouldDisplay(): boolean
    {
        return this.Command != null;
    }

    public static Deserialize(obj: any): LogModel
    {
        let trueObj: LogModel = new LogModel();
        trueObj.Type = obj.Type;
        trueObj.Process = obj.Process;
        trueObj.Command = CommandModel.Deserialize(obj.Command);

        return trueObj;
    }
}